controller/libpostal.js

```diff
@@ -114,12 +114,29 @@ function setup(libpostalService, should_execute) {
   return controller;
 }
 
+const RECAST_LABELS = [{ value: 'zoo', label: { to: 'house' } }];
 const DIAGONAL_DIRECTIONALS = ['ne','nw','se','sw'];
 
 // apply fixes for known bugs in libpostal
 function patchBuggyResponses(response){
   if( !Array.isArray(response) || !response.length ){ return response; }
 
+  // recast labels for certain values, currently only applied to parses which return a single label.
+  // the RECAST_LABELS array contains match/replace conditions which are applied in order.
+  // the 'value' and 'label.to' properties are mandatory, they define the value to match on and
+  // the replacement label to assign. you may optionally also provide 'label.from' which will restrict
+  // replacements to only records with BOTH a matching 'value' and a matching 'label.from'.
+  if( response.length === 1 ){
+    let first = response[0];
+    for( var ii=0; ii<RECAST_LABELS.length; ii++ ){
+      let recast = RECAST_LABELS[ii];
+      if( !recast.hasOwnProperty('label') || !recast.label.hasOwnProperty('to') ){ continue; }
+      if( recast.value !== first.value ){ continue; }
+      if( recast.label.hasOwnProperty('from') && recast.label.from !== first.label ){ continue; }
```

Comment:
This code is pretty defensive and hard to read. Some suggestions that might help:
- avoid use of `hasOwnProperty`, especially multiple times in a row to walk down a property chain. I've always disliked `hasOwnProperty` because it's such a lengthy function name (it's a shame it's so critical in Javascript). `_.get()` would probably work great here
- it's not clear whats different regarding the variable `first` vs `response[0]`. both are used, and `first` is not declared `const` so the expectation is it would be modified, but it's not
- RECAST_LABELS is not modified, so a `forEach` loop or something that provides more contextual hints and less looping boilerplate would help clarify what's going on over the for loop with generic iterator variable name

Reply:
I understand your reluctance to use `hasOwnProperty`, it's ugly and verbose, but it's the only mechanism which javascript provides for key-checking.

It's not analgous or correct to use a value check such as `!!recast.label` or `_.get()` because they do fundamentally different things.

So unfortunately we are stuck with the shitty language decisions of javascipt :P

I will change it to `_.has()` which saves a few extra keystrokes.

I favour using the core language features because I feel that despite them being shitty in javascript, are well known and predicable.

I think you favour using external libs to fix some of the ugly parts of js, and I'm cool with that, in the case of underscore I think it makes sense.

Reply:
Also worth having a look at the source code for `_.has()`, it's super complex and does a lot of things, in some cases it feels like overkill for a simple key-check.

Maybe there is a simpler solution for key-checking that we are not aware of?